### PR TITLE
Modeling rules fix

### PR DESCRIPTION
**Pull request by daryakoval**
fix modeling rules 
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5465
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5444

### CODE REVIEW

File: Tests/scripts/collect_tests/collect_tests.py [hunk 1]
```diff
@@ -31,7 +32,7 @@
 from Tests.scripts.collect_tests.utils import (ContentItem, Machine,
                                                PackManager, find_pack_folder,
                                                find_yml_content_type, to_tuple, hotfix_detect_old_script_yml,
-                                               FilesToCollect)
+                                               FilesToCollect, is_test_data_exist)
```
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:35]: please rename to `has_modeling_rule_test_data`

File: Tests/scripts/collect_tests/collect_tests.py [hunk 2]
```diff
@@ -580,8 +581,11 @@ def _collect_pack_for_modeling_rule(
             else:  # pragma: no cover
                 raise RuntimeError(f'Unexpected file type {file_type} for changed file {changed_file_path}')
         # the modeling rule to test will be the containing directory of the modeling rule's component files
-        relative_path_of_mr = PACK_MANAGER.relative_to_packs(changed_file_path)
-        modeling_rule_to_test = relative_path_of_mr.parent
+        modeling_rule_to_test = None
+        logging.info(f'hereee checking: {changed_file_path.parent}')
```
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:585]:
```suggestion
        logging.debug('checking {changed_file_path.parent}')
```

File: Tests/scripts/collect_tests/utils.py [hunk 3]
```diff
@@ -322,3 +324,10 @@ def hotfix_detect_old_script_yml(path: Path):
 class FilesToCollect(NamedTuple):
     changed_files: tuple[str, ...]
     pack_ids_files_were_removed_from: tuple[str, ...]
+
+
+def is_test_data_exist(modeling_rule_dir):
```
Comment (dorschw) [inline Tests/scripts/collect_tests/utils.py:329]: missing type hint, please use `Path` from pathlib

File: Tests/scripts/collect_tests/utils.py [hunk 4]
```diff
@@ -322,3 +324,10 @@ def hotfix_detect_old_script_yml(path: Path):
...
+    for modeling_rule_file in os.scandir(modeling_rule_dir):
+        if modeling_rule_file.name.endswith('testdata.json'):
+            return True
+    return False
```
Comment (dorschw) [inline Tests/scripts/collect_tests/utils.py:333]:
```suggestion
    return any((file.name.endswith('testdata.json') for file in modeling_rule_dir.glob("*")))
```

Comment (dorschw) [reply Tests/scripts/collect_tests/utils.py:333]: Also, please add a `if not modeling_rule_dir.is_dir(): raise ValueError("{modeling_rule_dir=} must be a directory") before the `return` statement

**Review by dantavori — APPROVED** [review]

**Pull request merged**